### PR TITLE
Add support for updating the connected git repository

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

Adds support for updating the connected git repository (GitHub is the only one supported as of now).

Usage:
```sh
tuist project update --repository https://github.com/tuist/tuist
```

The connected git repository is then shown as part of the `tuist project show tuist/tuist` command.

The connected git repository is used for:
- Automatically determining the default branch of a project (the default branch should be later synced with GitHub webhooks, not just queried when the repository is connected)
- Ensuring the Tuist project has permissions to interact with the GitHub project for the new GitHub app.

### How to test the changes locally 🧐

```
tuist project update --repository https://github.com/tuist/tuist
tuist project show tuist/tuist # the git repository should be printed in the output
```

Note these commands will work only once the server PR is merged. The new acceptance test will also fail until then.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistKit/Services/Project/ProjectUpdateService.swift [hunk 1]
```diff
@@ -55,7 +56,8 @@ struct ProjectUpdateService {
         _ = try await updateProjectService.updateProject(
             fullHandle: fullHandle,
             serverURL: serverURL,
-            defaultBranch: defaultBranch
+            defaultBranch: defaultBranch,
+            repositoryURL: repository
```
Comment (pepicrft) [inline Sources/TuistKit/Services/Project/ProjectUpdateService.swift:60]: The function's argument is called `repository`, but we pass it to the API as `repositoryURL`, so it feels a bit inconsistency. Should the API take a URL or a `tuist/tuist` slug? What's the most ergonomic option for developers? (thinking out loud)

File: Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift [hunk 2]
```diff
@@ -21,6 +21,11 @@ struct ProjectUpdateCommand: AsyncParsableCommand {
     )
     var defaultBranch: String?
 
+    @Option(
+        help: "Set the connected git repository. Example: --repository https://github.com/tuist/tuist"
+    )
+    var repository: String?
```
Comment (pepicrft) [inline Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift:27]: If we end up leaning on the API taking a URL I'd recommend renaming this to:
```suggestion
    var repositoryURL: String?
```

File: Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift [hunk 3]
```diff
@@ -21,6 +21,11 @@ struct ProjectUpdateCommand: AsyncParsableCommand {
     )
     var defaultBranch: String?
 
+    @Option(
+        help: "Set the connected git repository. Example: --repository https://github.com/tuist/tuist"
```
Comment (pepicrft) [inline Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift:25]:
```suggestion
        help: "Set the connected Git repository. Example: --repository https://github.com/tuist/tuist"
```

What happens if I pass `https://github.com/tuist/tuist.git`?

**Review by pepicrft — APPROVED** [review]

File: Sources/TuistKit/Services/Project/ProjectUpdateService.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/TuistKit/Services/Project/ProjectUpdateService.swift:60]: Changed it to `--repository-url`. I felt just `--repository` would be cleaner in the CLI context, but let's be more explicit and have the naming consistent with the API.

File: Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift (hunk 3 above)
Comment (fortmarek) [reply Sources/TuistKit/Commands/Project/ProjectUpdateCommand.swift:25]: That's also valid.

**Pull request merged**